platform/configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/ConfiguratorImpl.java

```diff
@@ -0,0 +1,425 @@
+    }
+
+    /**
+     * Stops the bundle with the given name.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String stopBundle(String bundleSymName) {
+        return registerHandler(BundleOperation.forStop(bundleSymName, getBundleContext()));
+    }
+
+    /**
+     * Determines if the bundle with the given name is started.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return true if started; else, false
+     */
+    public boolean isBundleStarted(String bundleSymName) {
+        try {
+            return BundleOperation.forStart(bundleSymName, getBundleContext())
+                    .readState();
+        } catch (ConfiguratorException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Installs and starts the feature with the given name.
+     *
+     * @param featureName the name of the feature
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String startFeature(String featureName) {
+        return registerHandler(FeatureOperation.forStart(featureName, getBundleContext()));
+    }
+
+    /**
+     * Stops the feature with the given name.
+     *
+     * @param featureName the name of the feature
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String stopFeature(String featureName) {
+        return registerHandler(FeatureOperation.forStop(featureName, getBundleContext()));
+    }
+
+    /**
+     * Determines if the feature with the given name is started.
+     *
+     * @param featureName the name of the feature
+     * @return true if started; else, false
+     */
+    public boolean isFeatureStarted(String featureName) {
+        return FeatureOperation.forStart(featureName, getBundleContext())
+                .readState();
+    }
+
+    /**
+     * Creates a property file in the system with the given set of new key:value pairs.
+     *
+     * @param propFile   the property file to create
+     * @param properties the set of key:value pairs to save to the property file
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String createPropertyFile(Path propFile, Map<String, String> properties) {
+        return registerHandler(PropertyOperation.forCreate(propFile, properties));
+    }
+
+    /**
+     * Deletes a property file in the system.
+     *
+     * @param propFile the property file to delete
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    public String deletePropertyFile(Path propFile) {
```

Comment:
Same question here about deleting. We should validate that they are trying to delete a property file and that it is within the bounds of the paths we deem acceptable.